uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Form/CreateUserRecordTaskForm.php

```diff
@@ -218,4 +224,19 @@ public function validateInput($fields) {
     return empty($errors) ? TRUE : $errors;
   }
 
+  /**
+   * Restores user account after delete
+   *
+   * @param object $user
+   * @param array $roles
+   * @param int $contactId
+   */
+  public function restoreAccount($user, $roles, $contactId) {
+    $this->drupalUserService->addRoles($user->mail, $roles);
+
+    $ufMatch['uf_id'] = $user->uid;
+    $ufMatch['contact_id'] = $contactId;
+    $ufMatch['uf_name'] = $user->mail;
+    CRM_Core_BAO_UFMatch::create($ufMatch);
```

Comment:
Would be better to use the Civi API here as it would be more reliable rather than a BAO method

Reply:
👍 I will change to API implementation.